### PR TITLE
Add s2e_is_forking_enabled api

### PR DESCRIPTION
**Pull request by emugnier**
See thread: https://groups.google.com/g/s2e-dev/c/dXkjjNaU28o/m/FTb3uhGRCwAJ

### CODE REVIEW

File: guest/common/include/s2e/s2e.h [hunk 1]
```diff
@@ -341,6 +341,18 @@ static inline void s2e_disable_forking(void)
     );
 }
 
+///
+/// \brief Check if forking is enabled on symbolic conditions
+///
+static inline int s2e_is_forking_enabled(void) {
+    int version;
```
Comment (vitalych) [inline guest/common/include/s2e/s2e.h:348]: Rename variable to something more appropriate

**Review by vitalych — CHANGES_REQUESTED** [review]

File: guest/common/include/s2e/s2e.h [hunk 2]
```diff
@@ -341,6 +341,18 @@ static inline void s2e_disable_forking(void)
...
+    __asm__ __volatile__(
+        S2E_INSTRUCTION_SIMPLE(BASE_S2E_IS_FORK_ENABLED)
+        : "=a" (version)  : "a" (0)
```
Comment (vitalych) [inline guest/common/include/s2e/s2e.h:351]: There is no need for input parameters

File: libs2eplugins/src/s2e/Plugins/Core/BaseInstructions.cpp [hunk 3]
```diff
@@ -797,6 +797,12 @@ void BaseInstructions::handleBuiltInOps(S2EExecutionState *state, uint64_t opcod
             break;
         }
 
+        case BASE_S2E_IS_FORK_ENABLED: { /* s2e_is_forking_enabled */
```
Comment (vitalych) [inline libs2eplugins/src/s2e/Plugins/Core/BaseInstructions.cpp:800]: IS_FORK and is_forking naming is inconsistent

**Comment by emugnier** [comment]
I fixed the typos as asked. Let me know if there is anything else I should do.

File: libs2eplugins/src/s2e/Plugins/Core/BaseInstructions.cpp (hunk 3 above)
Comment (vitalych) [reply libs2eplugins/src/s2e/Plugins/Core/BaseInstructions.cpp]: Use BASE_S2E_IS_FORKING_ENABLED for consistency

Comment (emugnier) [reply libs2eplugins/src/s2e/Plugins/Core/BaseInstructions.cpp]: Done

**Review by vitalych — APPROVED** [review]

**Pull request merged**